internal/reconciler/shoot/reconciler_test.go

```diff
@@ -149,6 +150,36 @@ var _ = Describe("Reconciler", func() {
 		))
 	})
 
+	It("should add trust-configurator shoot finalizer if missing", func() {
+		shoot.Finalizers = nil
+		Expect(fakeClient.Create(ctx, shoot)).To(Succeed())
+
+		res, err := reconciler.Reconcile(ctx, reconcile.Request{NamespacedName: shootObjectKey})
+		Expect(err).ToNot(HaveOccurred())
+		Expect(res).To(Equal(ctrl.Result{}))
+
+		var updatedShoot gardencorev1beta1.Shoot
+		Expect(fakeClient.Get(ctx, shootObjectKey, &updatedShoot)).To(Succeed())
+		Expect(updatedShoot.Finalizers).To(ContainElement(finalizer))
+	})
+
+	It("should do nothing when shoot trust-configurator finalizer is missing", func() {
```

Comment:
Do we need this test when we have the test "should delete OIDC resource because shoot is being deleted"?